enrichers/src/main/java/io/quarkus/qe/enricher/QuarkusExtensionsEnricher.java

```diff
@@ -57,26 +58,54 @@ private String getRawUrlFromRepository(Repository repository) throws EnrichmentE
                         repository.getRepoUrl()));
     }
 
-    private Set<QuarkusExtension> findAllDependencies(String baseUrl) throws EnrichmentException {
+    private Set<QuarkusExtension> findAllDependencies(String baseUrl, LookUpHelper lookUpHelper)
+            throws EnrichmentException {
         Set<QuarkusExtension> extensions = new HashSet<>();
 
         Model model = parseMavenModel(baseUrl);
-        extensions.addAll(model.getDependencies().stream().map(Dependency::getArtifactId)
-                .filter(artifact -> artifact.startsWith(QUARKUS_TAG))
-                .map(this::toQuarkusExtensionModel)
+        lookUpHelper.addAll(model.getProperties());
+        lookUpHelper.addAll(model.getDependencyManagement());
+        Optional.ofNullable(model.getBuild()).ifPresent(build -> {
+            lookUpHelper.addQuarkusMavenPlugin(build.getPluginManagement());
+            lookUpHelper.addQuarkusMavenPlugin(build.getPlugins());
+        });
```

Comment:
Can you refactor this block of code to just get the quarkus version and keep the internal details into the service?